12.0 mig sale_timesheet_invoice_description

ernestotejeda opened the pull request:
Cc @Tecnativa

carlosdauden committed:
[9.0][NEW] sale_timesheet_invoice_description: New module (#154)

[ADD] sale_timesheet_invoice_description: New module

oca-transbot committed:
OCA Transbot updated translations from Transifex

sbidoul committed:
[FIX] remove en.po that was erroneously created by transbot

oca-transbot committed:
OCA Transbot updated translations from Transifex

pedrobaeza committed:
[MIG] Make modules uninstallable

pedrobaeza committed:
[MIG] Rename manifest files

oca-transbot committed:
OCA Transbot updated translations from Transifex

pedrobaeza committed:
[MIG] sale_timesheet_invoice_description: Migration to 10.0

oca-transbot committed:
OCA Transbot updated translations from Transifex

astirpe committed:
[11.0][MIG] sale_timesheet_invoice_description: port to V11

oca-transbot committed:
OCA Transbot updated translations from Transifex

oca-travis committed:
[UPD] Update sale_timesheet_invoice_description.pot

pedrobaeza committed:
[FIX] sale_timesheet_invoice_description: Load en_US language in tests

pedrobaeza committed:
[FIX] sale_timesheet_invoice_description: Avoid new constraint

There's a constraint that limits the quantity over received one. Although it seems invalid
on services, we bypass it now with this.

rschnapka committed:
Translated using Weblate (German)

Currently translated at 100,0% (10 of 10 strings)

Translation: account-invoicing-11.0/account-invoicing-11.0-sale_timesheet_invoice_description
Translate-URL: https://translation.odoo-community.org/projects/account-invoicing-11-0/account-invoicing-11-0-sale_timesheet_invoice_description/de/

ernestotejeda committed:
[MIG] sale_timesheet_invoice_description: Migration to 12.0

pedrobaeza commented:
Please comment when Travis is green.

ernestotejeda committed:
[MIG] sale_timesheet_invoice_description: Fix E261 flake8 error

ernestotejeda commented:
Travis in green

pedrobaeza commented on sale_timesheet_invoice_description/models/sale.py:
```diff
@@ -29,7 +29,12 @@ class SaleOrderLine(models.Model):
-            details.append(line.date)
+            # If project_timesheet_time_control module is installed,
+            # date field is hidden and the new date_time field is used instead
+            if hasattr(line, 'date_time'): # pragma: no cover
```
Although the field is hidden, `date` field is still synchronized, so I prefer to keep the previous behavior showing only date. Customer doesn't need this kind of detail.

ernestotejeda replied:
Yes, I agree, thanks

pedrobaeza commented on sale_timesheet_invoice_description/models/sale.py:
```diff
@@ -47,9 +52,14 @@ def _prepare_invoice_line(self, qty):
+        # date field is hidden and the new date_time field is used instead
+        line_obj = self.env['account.analytic.line']
+        order = hasattr(line_obj, 'date_time') and 'date_time' or 'date'
+        for line in line_obj.search(domain, order=order + ', id'):
```
Why don't rely in default order set by the model instead (and more now that we are going to set it properly)?

ernestotejeda replied:
I think you're right, thank you

ernestotejeda committed:
[MIG] sale_timesheet_invoice_description: Improve code

Keep the previous behavior showing only date
Remove order parameter in sear method

rafaelbn reviewed (approved):
Tested in runbot :+1:
